### ISSUE TITLE
Prepare data for quantitative analysis

### ISSUE THREAD

**Issue by nlitz88** [nlitz88/workzone#21]
- [x] Need to create a rosbag recording the entire pipeline running. This will capture both the intermediate BEVFusion output AND the BEV map with the predicted workzone masks. DONE: Here: https://drive.google.com/file/d/1J2QiT3MSH1UXkZMs6GTJpIXaqSVx7RnP/view?usp=drive_link (shouldn't need to use this, though).
- [x] Will find some kind of utility to export / extract the images from the ros bags for @brendaju to use.
- [x] Upload the images to Google Drive. DONE: Here: https://drive.google.com/file/d/1w5oQI9uWOR0VSK9zqobMvtuqYPXzDHIq/view?usp=drive_link

**Comment by nlitz88** [comment]
@brendaju Okay, the images are linked above in that third bullet point. Let me know if you are able to access them okay.

**Comment by nlitz88** [comment]
WAIT--There is one potential problem with these: **I don't think we can trust the numbers/indexes of those images**. The export tool I used (https://github.com/MapIV/ros2_bag_to_image) had an issue where if your image messages weren't timestamped, then it would bug out--so I had to increment things manually. However, this could lead to some inconsistencies.

To resolve this, the *right* answer is to properly timestamp our image messages. Not rocket science, but would just probably be a tiny bit of leg work from @CMUBOB97 and myself.

@CMUBOB97 I can look myself, but before I dive in: do you know where the BEVFusion node creates the `image_msg` published to `result_image`? I.e., where we _might be able to set the timestamp?_

Basically, we just need to set the timestamp in the BEVFusion node. THEN, we have to update Chi's node to **set the timestamp of the `boundary_image` we create to be == the timestamp found in the received result_image's header*

**Comment by nlitz88** [comment]
@CMUBOB97 Update: I just found where they create the image and it looks like they didn't set the timestamp. I just added a fix here in the `ros2-galactic` branch I made yesterday https://github.com/nlitz88/BEVFusion-ROS-TensorRT/commit/238103446b2834e32b3df733e1d80ce9725586f8, so hopefully we are good/set on the BEVFusion node front.

Now all we need is a quick one-line update to the boundary generation node.

**Comment by nlitz88** [comment]
@CMUBOB97 Another update: I just updated your node so that the published image's timestamp matches the timestamp of the incoming message. https://github.com/CMUBOB97/ROS2-Boundary-Segmentation/commit/18f3c47e3b27c20b8e80b0d3c4cfe2f2ab5988dc

So in theory, _now_ we should be okay. I.e., now we can know which `result_image` from BEVFusion truly corresponds to each `boundary_image` from the boundary segmentation node.

I will re-record those ROS bags and upload them again. Sorry for the delay.

**Comment by nlitz88** [comment]
Okay, re-recorded, exported, and updated the google drive links above. For now, just have images from Scene 61, but I can make some additional scenes as well.

**Comment by CMUBOB97** [comment]
Yay!

**Comment by CMUBOB97** [comment]
@nlitz88 I just pushed a change on boundary segmentation package so that it overlays the ego frame on top of the segmented workzone (boundary_image). Also let me know if you need help in generating the binary prediction results (or maybe you have already acquired it).

**Comment by nlitz88** [comment]
@CMUBOB97 

![Image](https://github.com/nlitz88/workzone/assets/9357423/d047e671-333c-43ad-a401-1dab90c5f54a)

Okay, I see the ego-vehicle overlaid. Very cool!
